### ISSUE TITLE
Introduce citus_invalidate_node_cache

### ISSUE THREAD

**Issue by JelteF** [citusdata/citus#6558]
For DNS based failovers where citus_update_node is not called it would be useful if it was possible to invalidate the connection caches for a specific node. This is also useful in combination with `citus_pause_node` which is proposed in #6557 , to invalidate the connection caches after a coordinated restart.

**Comment by marcocitus** [comment]
I don't think we need this is really necessary after https://github.com/citusdata/citus/pull/6404
